partner_search_alias/models/res_partner.py

```diff
@@ -12,3 +14,19 @@ class ResPartner(models.Model):
     @property
     def _rec_names_search(self):
         return list(set(super()._rec_names_search + ["search_alias"]))
+
+    @api.model
+    def get_view(self, view_id=None, view_type="form", **options):
+        res = super().get_view(view_id, view_type, **options)
+        if view_type == "search":
+            xml = etree.XML(res["arch"])
+            name_field = xml.xpath("//field[@name='name']")
+            if name_field:
+                name = name_field[0]
+                filter_domain = name.get("filter_domain", "[]")
+                filter_domain = filter_domain.replace(
+                    "[", "['|', ('search_alias', 'ilike', self), "
+                )
```

Comment:
Shouldn't we be using `expression.OR()` instead?

Reply:
To use` expression.OR()`, the XML string must be converted to a Python list using `safe_eval`. However, `safe_eval` fails if the string includes unsupported expressions like `self`. Seem like we have to replace `self` with some values and I think we can't do it. So, I prefer using current approach. How do you think?

Reply:
I think we can leave the code as is on second thought. Thanks for checking.